src/core/Akka.Remote/Remoting.cs

```diff
@@ -186,35 +186,29 @@ public override void Start()
 
                 try
                 {
-                    var addressPromise = new TaskCompletionSource<IList<ProtocolTransportAddressPair>>();
+                    var addressPromise = new TaskCompletionSource<IList<ProtocolTransportAddressPair>>(TaskCreationOptions.RunContinuationsAsynchronously);
 
                     // tells the EndpointManager to start all transports and bind them to listenable addresses, and then set the results
                     // of this promise to include them.
-                    _endpointManager.Tell(new EndpointManager.Listen(addressPromise)); 
+                    _endpointManager.Tell(new EndpointManager.Listen(addressPromise));
 
                     addressPromise.Task.Wait(Provider.RemoteSettings.StartupTimeout);
                     var akkaProtocolTransports = addressPromise.Task.Result;
-                    if(akkaProtocolTransports.Count==0)
+                    if (akkaProtocolTransports.Count == 0)
                         throw new ConfigurationException(@"No transports enabled under ""akka.remote.enabled-transports""");
-                    _addresses = new HashSet<Address>(akkaProtocolTransports.Select(a => a.Address));
 
-                    IEnumerable<IGrouping<string, ProtocolTransportAddressPair>> tmp =
-                        akkaProtocolTransports.GroupBy(t => t.ProtocolTransport.SchemeIdentifier);
-                    _transportMapping = new Dictionary<string, HashSet<ProtocolTransportAddressPair>>();
-                    foreach (var g in tmp)
-                    {
-                        var set = new HashSet<ProtocolTransportAddressPair>(g);
-                        _transportMapping.Add(g.Key, set);
-                    }
+                    var transportMapping = new Dictionary<string, HashSet<ProtocolTransportAddressPair>>();
+                    foreach (var g in akkaProtocolTransports.GroupBy(t => t.ProtocolTransport.SchemeIdentifier))
+                        transportMapping.Add(g.Key, new HashSet<ProtocolTransportAddressPair>(g));
```

Comment:
Nice simplification. Should we .As/ToDictionary() to go further?

Reply:
will try the key point was that it set the dictionary and filled it up and not set the filled directory to the public field